django_emarsys/models.py

```diff
@@ -141,12 +76,63 @@ class EventInstance(models.Model):
     state = models.CharField(max_length=1024, choices=STATE_CHOICES,
                              default='sending')
 
+    def __str__(self):
+        return 'at {}: {}'.format(self.when, self.event.name)
+
+
+# deprecated
+class EventInstanceData(models.Model):
+    event_trigger = models.ForeignKey(EventInstance)
+    event_data = models.ForeignKey(EventData)
+    content_type = models.ForeignKey(ContentType)
+    object_id = models.PositiveIntegerField()
+    content_object = GenericForeignKey('content_type', 'object_id')
+
+    class Meta:
+        unique_together = [('event_trigger', 'event_data')]
+
+
+# new classes
+
+class EmarsysEventId(models.Model):
+    event_name = models.CharField(max_length=1024, primary_key=True)
+    emarsys_id = models.IntegerField()
+
+    objects = managers.EmarsysEventIdManager()
+
+
+@python_2_unicode_compatible
+class NewEvent(models.Model):
+    STATE_CHOICES = [('sending', 'sending'),
+                     ('error', 'error'),
+                     ('success', 'success')]
+
+    SOURCE_CHOICES = [('automatic', 'automatic'),
+                      ('manual', 'manual')]
+
+    event_name = models.CharField(max_length=1024)
+    user = models.ForeignKey(settings.AUTH_USER_MODEL, null=True)
+    recipient_email = models.CharField(max_length=1024)
+    context = JSONField(null=True)
+    data = JSONField()
```

Comment:
Ah, ok, jsonfield is the one I'm most familiar with. I should've checked the rest of the code first.